pymc3/step_methods/metropolis.py

```diff
@@ -154,14 +212,20 @@ def __init__(self, vars=None, S=None, proposal_dist=None, scaling=1.,
 
         self.mode = mode
 
+        # flag to indicate this stepper was instantiated within an MLDA stepper
+        # if not, tuning parameters are reset when _iter_sample() is called
+        self.is_mlda_base = kwargs.pop("is_mlda_base", False)
+
         shared = pm.make_shared_replacements(vars, model)
         self.delta_logp = delta_logp(model.logpt, vars, shared)
         super().__init__(vars, shared)
 
     def reset_tuning(self):
-        """Resets the tuned sampler parameters to their initial values."""
-        for attr, initial_value in self._untuned_settings.items():
-            setattr(self, attr, initial_value)
+        """Resets the tuned sampler parameters to their initial values.
```

Comment:
I think it would be cleaner to inherit `Metropolis` to something like `MetropolisMLDA` or something and overwrite these methods rather than change the base-class in this way.

The more isolated the code changes are from existing code the better. In principle, adding a new sampler should not require code changes to existing ones. That way everything stays nice and modular.

Reply:
Yes, I was uncertain of the best way to implement this, an inherited class makes the most sense indeed.

Reply:
New class MetropolisMLDA added to mlda.py